Projects/Backendless.NET.Standard20/NewHttpEngine.cs

```diff
@@ -0,0 +1,226 @@
+﻿using System;
+using System.Collections;
+using System.Collections.Generic;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+using Weborb.V3Types;
+using System.Net.Http;
+using Weborb.Message;
+using Weborb.Util;
+using Weborb.Reader;
+using System.Net;
+using System.Net.Http.Headers;
+using Weborb.Types;
+using Weborb.Protocols.Amf;
+using Weborb.Exceptions;
+
+namespace Weborb.Client
+{
+  public class NewHttpEngine : Engine
+  {
+    HttpClient httpClient = new HttpClient();
+    HttpWebRequest _request;
+    public NewHttpEngine( String url, IdInfo idInfo ) : base( url, idInfo )
+    {
+    }
+
+    public override void SendRequest<T>( V3Message v3Msg, IDictionary requestHeaders,
+                                         IDictionary httpHeaders, Responder<T> responder,
+                                         AsyncStreamSetInfo<T> asyncStreamSetInfo ) =>
+      ThreadPool.QueueUserWorkItem( state => SendHttpRequest( v3Msg, requestHeaders, httpHeaders, responder, asyncStreamSetInfo ) );
```

Comment:
why do you use threadpool here?

Reply:
Because if the threadpool is full, you have to wait until it is free. Otherwise it can lead to errors.